docs/development/design/refactor-remote-repository.rst

```diff
@@ -0,0 +1,101 @@
+======================================
+ Refactor ``RemoteRepository`` object
+======================================
+
+
+This document describes the current usage of ``RemoteRepository`` objects and proposes a new normalized modeling.
+
+
+Goals
+=====
+
+* De-duplicate data stored in our database.
+* Save only one ``RemoteRepository`` per GitHub repository.
+* Use an intermediate table between ``RemoteRepository`` and ``User`` to store associated remote data for the specific user.
+* Make this model usable from our SSO implementation.
+* Use Post ``JSONField`` to store associated ``json`` remote data.
```

Comment:
This could be considered a "Nice to have" in case we don't have the time to make all the changes required to depend on PostgreSQL in tests. If we keep the `admin=` field in the relationship, filtering over the JSON is not needed.

Reply:
I think we can probably wait here until we migrate to Django 3.x that implements this natively and avoid making this migration twice. We don't really need  json field right now.